CRM/Pledge/BAO/Query.php

```diff
@@ -405,7 +405,7 @@ public static function from($name, $mode, $side) {
         break;
 
       case 'pledge_status':
-        $from .= " $side JOIN civicrm_option_group option_group_pledge_status ON (option_group_pledge_status.name = 'contribution_status')";
+        $from .= " $side JOIN civicrm_option_group option_group_pledge_status ON (option_group_pledge_status.name = {$name})";
```

Comment:
`$name` is always `pledge_status`. But, just for readability, I'll update this to text instead of variable.